File: AccountingApp.API/Models/Product.cs
using System;
using System.Collections.Generic;

namespace AccountingApp.API.Models
{
    // 1: Physical (Stock Tracking), 2: Service (No Stock Tracking)
    public enum ProductType : byte
    {
        Physical = 1,
        Service = 2
    }

    public class Product : BaseEntity<int>
    {
        public int CategoryId { get; set; }
        public Category Category { get; set; } = null!;

        public string Code { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string SerialNumber { get; set; } = string.Empty; // Seri No
        public string Barcode { get; set; } = string.Empty;

        public decimal UnitPrice { get; set; }
        
        public int CurrencyId { get; set; }
        public Currency Currency { get; set; } = null!;

        public ProductType Type { get; set; }
        
        public decimal StockQuantity { get; set; } // Can be 0 for services

        public ICollection<StockMovement> StockMovements { get; set; } = new List<StockMovement>();
    }
}

File: accounting_app_ui/lib/models/product.dart
class Product {
  final int id;
  final String code;
  final String name;
  final String? description;
  final String? serialNumber;
  final String? barcode;
  final double unitPrice;
  final int type; // 1: Physical, 2: Service
  final double stockQuantity;
  final int categoryId;
  final String categoryName;
  final int currencyId;
  final String currencyCode;
  final String currencySymbol;
  final bool isActive;

  Product({
    required this.id,
    required this.code,
    required this.name,
    this.description,
    this.serialNumber,
    this.barcode,
    required this.unitPrice,
    required this.type,
    required this.stockQuantity,
    required this.categoryId,
    required this.categoryName,
    required this.currencyId,
    required this.currencyCode,
    required this.currencySymbol,
    required this.isActive,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      code: json['code'] ?? '',
      name: json['name'] ?? '',
      description: json['description'],
      serialNumber: json['serialNumber'],
      barcode: json['barcode'],
      unitPrice: (json['unitPrice'] ?? 0).toDouble(),
      type: json['type'] ?? 1,
      stockQuantity: (json['stockQuantity'] ?? 0).toDouble(),
      categoryId: json['categoryId'] ?? 0,
      categoryName: json['categoryName'] ?? '',
      currencyId: json['currencyId'] ?? 1,
      currencyCode: json['currencyCode'] ?? 'TRY',
      currencySymbol: json['currencySymbol'] ?? '₺',
      isActive: json['isActive'] ?? true,
    );
  }

  String get typeText => type == 1 ? 'Fiziksel' : 'Hizmet';
  bool get isPhysical => type == 1;
}
